### PR TITLE
Add stylelint support

### PR DESCRIPTION
**Pull request by XhmikosR**
@REJack this should supersede the https://github.com/ColorlibHQ/AdminLTE/tree/feat/stylelint branch.

I didn't add `fusv` because you first need to address #2725.

Waiting for #2721 and then I'll add `npm test` on CI too.

PS. better not squash the patches.
PPS. These changes will probably slightly improve gzip compression, at least for the non-minified CSS files. We use bundlewatch to monitor the sizes, feel free to add it or let me know and I could take a stab at it after there is a working CI.

Non-whitespace diff: https://github.com/ColorlibHQ/AdminLTE/pull/2727/files?w=1

### CODE REVIEW

File: .stylelintignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+**/*.min.css
```
Comment (REJack) [inline .stylelintignore:1]: Why we ignore not any css files? I would only lint the scss files. 🤣

Comment (XhmikosR) [reply .stylelintignore:1]: Well, I leave the filetype unspecified. It will lint whatever files are supported. We do the same upstream too.

**Comment by XhmikosR** [comment]
@REJack how do you want the CI to be? Running `npm test` on each matrix seems redundant so I was thinking to only run it on Ubuntu and Node.js 12.x.

**Comment by REJack** [comment]
Yup `npm test` should only executed one time, not like the Node.js Install test 😄 ubuntu-latest with Node.js 12 sounds good for me.

File: .stylelintignore (hunk 1 above)
Comment (REJack) [reply .stylelintignore:1]: Ok :D it was only a question :P

**Comment by XhmikosR** [comment]
Actually, now that I think about it I could move lint to a separate workflow. I think it makes more sense. And keep ci.yml with deps + building.

**Comment by REJack** [comment]
Yeah I would create seperate workflow for linting too.

**Comment by XhmikosR** [comment]
@REJack WDYT?

**Comment by XhmikosR** [comment]
BTW you really need to remove `install` because it runs on every `npm i`. See #2722.

File: .stylelintignore (hunk 1 above)
Comment (XhmikosR) [reply .stylelintignore:1]: No worries :)

File: .stylelintrc [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
+  "extends": [
+    "stylelint-config-twbs-bootstrap/scss"
+  ],
+  "rules": {
+    "declaration-no-important": null,
+    "order/properties-order": null,
```
Comment (XhmikosR) [inline .stylelintrc:7]: I set all these to `null` but should be revisited later, which should help with #2742

**Comment by XhmikosR** [comment]
Anyway feel free to merge @REJack if it looks good to you. But better not squash the patches :)

**Comment by REJack** [comment]
I look later over this and the other PRs & Issues, I'm not at my Notebook right now 😄

**Comment by XhmikosR** [comment]
I might need to adapt this after #2716 is merged, so I'll mark it as draft until that's merged.

File: .stylelintrc (hunk 2 above)
Comment (REJack) [reply .stylelintrc:7]: > I set all these to null but should be revisited later, which should help with #2742.

I've added a note to the v3.1 Project to keep an eye on it 😄

**Comment by XhmikosR** [comment]
@REJack please don't squash the patches when I say so :/ It's better for history to keep some patches separate.

This applies to this PR too.

**Comment by REJack** [comment]
Should I merge my branch before merging this PR or it is not needed?

**Comment by XhmikosR** [comment]
No, I can always fix merge conflicts and rebase myself. :)

**Pull request merged**